aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/AuthenticationState.kt

```diff
@@ -150,11 +150,15 @@ sealed class AuthenticationState : State {
                     }
                 }
                 is FederatedToIdentityPool -> {
-                    when (authenticationEvent) {
-                        is AuthenticationEvent.EventType.SignOutRequested -> {
+                    val authorizationEvent = event.isAuthorizationEvent()
+                    when {
+                        authenticationEvent is AuthenticationEvent.EventType.SignOutRequested -> {
                             val action = authenticationActions.initiateSignOutAction(authenticationEvent, null)
                             StateResolution(SigningOut(), listOf(action))
                         }
+                        authorizationEvent is AuthorizationEvent.EventType.StartFederationToIdentityPool -> {
```

Comment:
Just to confirm, is this for refresh?

Reply:
Refresh or replace. It could be an entirely different provider, or it could be same provider with different token.